src/pages/index.js

```diff
@@ -2,86 +2,166 @@
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    const { Link: link, Title: title } = inputFieldValues;
-    createCard({ name: title, link: link });
+    api
+      .uploadCard(inputFieldValues)
+      .then((res) => {
+        cardsListSection.addItem(
+          createCard({ name: inputFieldValues.Title, link: inputFieldValues.Link, _id: res._id }),
+          false
+        );
+      })
+      .catch((err) => alert(err))
+      .finally(() => addCardPopup.closeAfterSubmit());
   },
   addFormValidator
 );
-addImagePopup.setEventListeners();
+addCardPopup.setEventListeners();
+
+api
+  .getInitialCards()
+  .then((res) => cardsListSection.renderItems(res, true))
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => userInfo.setUserInfo(res))
+  .catch((err) => alert(err));
 
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardsList = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      createCard(item);
-    },
-  },
-  cardsListSection
-);
-cardsList.renderItems();
+const confirmPopup = new PopupConfirm("#confirm-modal", (cardData, evt) => {
+  evt.preventDefault();
+  api
+    .deleteCard(cardData.cardId)
+    .then(() => {
+      cardData.cardEl.remove();
+      cardData.cardEl = null;
+    })
+    .catch((err) => alert(err));
+});
+confirmPopup.setEventListeners();
 
 //add event listeners
-profileEditButton.addEventListener("click", openProfileForm);
-profileAddButton.addEventListener("click", openAddForm);
-
-//event listener callbacks
-function openProfileForm() {
+profileEditButton.addEventListener("click", () => {
   profilePopup.open();
-  const { name, title } = userInfo.getUserInfo();
-  profileModalNameInput.value = name;
-  profileModalDescInput.value = title;
+  profileModalNameInput.value = userInfo.getUserInfo().name;
+  profileModalDescInput.value = userInfo.getUserInfo().about;
   profileFormValidator.resetValidation();
   profileFormValidator.toggleButtonState();
-}
-
-function openAddForm() {
-  addImagePopup.open();
-}
+});
+profileAddButton.addEventListener("click", () => addCardPopup.open());
+profileAvatarEdit.addEventListener("click", () => avatarEditPopup.open());
 
+//functions
 function createCard(card) {
-  const newCard = new Card(card, "#card-template", () => {
-    imagePopup.open(newCard);
-  });
-  cardsList.addItem(newCard.generateCardElement());
+  const newCard = new Card(
+    card,
+    "#card-template",
+    () => {
+      imagePopup.open(newCard);
+    },
+    (cardData) => {
+      confirmPopup.open(cardData);
+    },
+    (card, method) => {
+      api
+        .updateCardLike(card.id, method)
+        .then((res) => {
+          if (res.isLiked) {
+            card.cardLikeIcon.classList.add("card__like-button_pressed");
+            card.isLiked = true;
+          } else {
+            card.cardLikeIcon.classList.remove("card__like-button_pressed");
+            card.isLiked = false;
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle